src/main/java/com/example/buyingCurrencyService/handlers/exception/NoConnectionWithServiceException.java

```diff
@@ -0,0 +1,19 @@
+package com.example.buyingCurrencyService.handlers.exception;
+
+public class NoConnectionWithServiceException extends RuntimeException{
```

Comment:
would be better to create a common application exception to build hierarchy. and after that extend it.